plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XWorkspaceManager.java

```diff
@@ -137,6 +171,43 @@ public boolean hasContent(URI uri) {
 		}
 	};
 
+	/**
+	 * Tells whether the workspace is in dirty state. The workspace is said to be in dirty state iff at least one file
+	 * is open AND is dirty, i.e. has unsaved changes.
+	 */
+	public boolean isDirty() {
```

Comment:
Can we call this `hasDirtyDocuments()`

Reply:
It was a deliberate decision of Marcus and me to introduce this term also on the workspace level (simplifies talking about various special cases of handling theses states).